feature/auth/src/commonTest/kotlin/org/mifospay/feature/auth/login/LoginViewModelTest.kt

```diff
@@ -0,0 +1,185 @@
+import kotlinx.coroutines.test.TestScope
+import kotlinx.coroutines.test.advanceUntilIdle
+import kotlinx.coroutines.test.resetMain
+import kotlinx.coroutines.test.runTest
+import kotlinx.coroutines.test.setMain
+import org.mifospay.core.common.DataState
+import org.mifospay.core.data.repository.ClientRepository
+import org.mifospay.core.data.util.Constants.INVALID_CREDENTIALS
+import org.mifospay.core.datastore.UserPreferencesRepository
+import org.mifospay.core.domain.LoginUseCase
+import org.mifospay.feature.auth.fakes.FakeAuthenticationRepository
+import org.mifospay.feature.auth.fakes.fakeClient
+import org.mifospay.feature.auth.login.LoginAction
+import org.mifospay.feature.auth.login.LoginState
+import org.mifospay.feature.auth.login.LoginViewModel
+import kotlin.test.AfterTest
+import kotlin.test.BeforeTest
+import kotlin.test.Test
+import kotlin.test.assertEquals
+import kotlin.test.assertFalse
+import kotlin.test.assertIs
+import kotlin.test.assertNull
+import kotlin.test.assertTrue
+
+@OptIn(ExperimentalCoroutinesApi::class)
+class LoginViewModelTest {
+    private val testDispatcher: CoroutineDispatcher = StandardTestDispatcher()
+    private val testScope = TestScope(testDispatcher)
+
+    private lateinit var viewModel: LoginViewModel
+
+    private val mockClientRepository = mock<ClientRepository> {
+        everySuspend { getClient(any()) } returns DataState.Success(fakeClient)
+    }
+
+    private val mockUserPreferencesRepository = mock<UserPreferencesRepository> {
+        everySuspend { updateToken(any()) } returns DataState.Success(Unit)
+        everySuspend { updateClientInfo(any()) } returns DataState.Success(Unit)
+        everySuspend { updateUserInfo(any()) } returns DataState.Success(Unit)
+    }
+
+    private var fakeAuthRepository = FakeAuthenticationRepository(shouldSucceed = true)
+    private lateinit var loginUseCase: LoginUseCase
+
+    @BeforeTest
+    fun setUp() {
+        Dispatchers.setMain(testDispatcher)
+
+        loginUseCase = LoginUseCase(
+            repository = fakeAuthRepository,
+            ioDispatcher = testDispatcher,
+            clientRepository = mockClientRepository,
+            userPreferencesRepository = mockUserPreferencesRepository,
+        )
+
+        viewModel = LoginViewModel(
+            loginUseCase = loginUseCase,
+            savedStateHandle = SavedStateHandle(),
+        )
+    }
+
+    @AfterTest
+    fun tearDown() {
+        Dispatchers.resetMain()
+    }
+
+    @Test
+    fun loginViewModel_InitialState_ValidInitialConditions(): TestResult {
+        return runTest(testDispatcher) {
+            assertEquals("", viewModel.stateFlow.value.username)
+            assertEquals("", viewModel.stateFlow.value.password)
+            assertFalse(viewModel.stateFlow.value.isPasswordVisible)
+            assertEquals(null, viewModel.stateFlow.value.dialogState)
+        }
+    }
+
+    @Test
+    fun `username change updates state`() = runTest(testDispatcher) {
```

Comment:
use one style for naming the test. 
for example `loginViewModel_UsernameChanged_UpdatesState`